Frameworks/Foundation/NSURLResponse.mm

```diff
@@ -89,8 +94,22 @@ - (id)_initWithURL:(NSURL*)url
         _url.attach([url copy]);
         _mimeType.attach([mimeType copy]);
         _textEncodingName.attach([textEncodingName copy]);
-        if (nil != fileName) {
-            _suggestedFilename.attach([fileName copy]);
+
+        if (fileName && ![fileName isEqualToString:s_unknownFileName]) {
+            woc::unique_cf<CFStringRef> identifier {
+                _UTTypeCreatePreferredIdentifierForTag(static_cast<CFStringRef>(@"public.mime-type"),
+                                                       static_cast<CFStringRef>(mimeType),
+                                                       nullptr)
+            };
+            StrongId<NSString> extension {
+                static_cast<NSString*>(
+                    _UTTypeCopyPreferredTagWithClass(identifier.get(), static_cast<CFStringRef>(@"public.filename-extension")))
+            };
+
+            // If the filename already has the extension do not append
+            _suggestedFilename.attach((extension && ![extension isEqualToString:[fileName pathExtension]]) ?
```

Comment:
this case ensures that if something has foo.pdf and the designated extension is .txt , it will get appended,
but if the designated extension is .pdf the it won't

---
In reply to: [95272900](https://github.com/Microsoft/WinObjC/pull/1662#discussion_r95272900) [](ancestors = 95272900)